crates/network-scanner/src/mdns.rs

```diff
@@ -19,10 +19,34 @@ impl MdnsDaemon {
     pub fn get_service_daemon(&self) -> mdns_sd::ServiceDaemon {
         self.service_daemon.clone()
     }
+
+    pub fn stop(&self) {
+        let receiver = match self.service_daemon.shutdown() {
+            Ok(receiver) => receiver,
+            Err(e) => {
+                // if e is try again, we should try again, but only once
+                let result = if matches!(e, mdns_sd::Error::Again) {
+                    self.service_daemon.shutdown()
+                } else {
+                    tracing::error!(error = %e, "Failed to shutdown service daemon");
+                    Err(e)
+                };
+
+                let Ok(receiver) = result else {
+                    warn!("Failed to shutdown service daemon");
+                    return;
+                };
+
+                receiver
+            }
+        };
+
+        let _ = receiver.recv_timeout(std::time::Duration::from_millis(100));
```

Comment:
Can you add a comment explaining the purpose of this?

Reply:
added